listen/map-changes.js

```diff
@@ -85,39 +110,41 @@ MapChanges.prototype.removeMapChangeListener = function (listener, token, before
         listeners = descriptor.changeListeners;
     }
 
-    var node = listeners.findLast(listener);
-    if (!node) {
+    var index = listeners.lastIndexOf(listener);
+    if (index === -1) {
         throw new Error("Can't remove map change listener: does not exist: token " + JSON.stringify(token));
     }
-    node["delete"]();
+    listeners.splice(index, 1);
 };
 
 MapChanges.prototype.dispatchMapChange = function (key, value, beforeChange) {
-    var descriptors = this.getAllMapChangeDescriptors();
-    var changeName = "Map" + (beforeChange ? "WillChange" : "Change");
-    descriptors.forEach(function (descriptor, token) {
+    var descriptors = this.getAllMapChangeDescriptors(),
+        changeName = "Map" + (beforeChange ? "WillChange" : "Change"),
+        listeners, tokenName, i, token, descriptor;
 
+    for (token in descriptors) {
+		descriptor = descriptors[token];
         if (descriptor.isActive) {
             return;
         } else {
             descriptor.isActive = true;
         }
 
-        var listeners;
         if (beforeChange) {
             listeners = descriptor.willChangeListeners;
         } else {
             listeners = descriptor.changeListeners;
         }
 
-        var tokenName = "handle" + (
+        tokenName = "handle" + (
             token.slice(0, 1).toUpperCase() +
             token.slice(1)
         ) + changeName;
 
         try {
             // dispatch to each listener
-            listeners.forEach(function (listener) {
+            var listener;
+            for(i=0;(listener = listeners[i]);i++) {
```

Comment:
Inner parentheses are superfluous. Please pad with spaces to be consistent with the surrounding style.